torch/csrc/jit/tensorexpr/mem_dependency_checker.cpp

```diff
@@ -64,8 +64,6 @@ std::vector<const Expr*> AccessInfo::getIndices() const {
       indices = load->indices();
     } else if (auto* call = dynamic_cast<const FunctionCall*>(expr_)) {
       indices = call->params();
-    } else if (auto* reduce = dynamic_cast<const ReduceOp*>(expr_)) {
```

Comment:
This is assuming that the dependency analysis does not need to happen at the Tensor level. Is that always guaranteed now? I understand that, if we do the cleanup as @ZolotukhinM suggested, then we are going to end up there. But if we use this dependency analysis before lowering to stmt now, I think we will need this. Am I missing something?

Reply:
It looks like loads and stores are present even before lowering ReduceOp completely. So, this should be fine AFAICT.